test/spmd/test_xla_sharding.py

```diff
@@ -318,78 +317,21 @@ def test_mark_sharding_not_ordered_partial_3d(self):
     # Shard along two axes if four or more devices are available
     z_dim = 2 if self.n_devices >= 4 else 1
     mesh = self._get_mesh((z_dim, 1, self.n_devices // z_dim))
-
-    # Expect local shard size to be [8, 16 / z_dim, 32]
-    xt1 = xs.mark_sharding(t1, mesh, (1, 0, None))
-
-    for local_shard in xt1.local_shards:
-      self.assertEqual(local_shard.data.size()[0], 8)
```

Comment:
any reason we removed the local shard size check here?

Reply:
Thanks, I am going to work with @khatwanimohit @jonb377 to add/refactor the tests.

Reply:
@JackCaoG actually, just realized that this was replaced by the assertion in L334